x-pack/plugins/ingest_pipelines/public/url_generator.test.ts

```diff
@@ -0,0 +1,108 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { IngestPipelinesUrlGenerator } from './url_generator';
+import { INGEST_PIPELINES_PAGES } from './application/services/navigation';
+
+describe('IngestPipelinesUrlGenerator', () => {
+  const getAppBasePath = (absolute: boolean = false) => {
+    if (absolute) {
+      return Promise.resolve('http://localhost/app/test_app');
+    }
+    return Promise.resolve('/app/test_app');
+  };
+  const urlGenerator = new IngestPipelinesUrlGenerator(getAppBasePath);
+
+  describe('Pipelines List', () => {
+    it('should generate correct relative url for list without pipelineId', async () => {
```

Comment:
This is just a pet peeve of mine so feel free to disregard it. 😄   I noticed that every description contains the phrase `should generate correct <result>` and I think the words "should" and "correct' become a bit redundant for test descriptions. If you wanted to make them terser I think you could rephrase them as `generates <result>`.